etest_test/build_test.py

```diff
@@ -1,12 +1,236 @@
 """Tests for etest.build."""
 
+import logging
+import textwrap
+
 import click.testing
+import docker
+import pytest
 
 import etest.build as sut
+from etest import profile
+
+logger = logging.getLogger(__name__)
 
 
 def test_help() -> None:
     """Ensure etest-build --help exits successfully."""
     runner = click.testing.CliRunner()
     result = runner.invoke(sut.main, ["--help"])
+
+    assert result.exit_code == 0
+
+
+def test_nobuild() -> None:
+    """Ensure etest-build outputs 4 info lines when not building."""
+    runner = click.testing.CliRunner()
+    result = runner.invoke(sut.main, ["--nobuild", "-v", "INFO"])
+
+    assert result.exit_code == 0
+
+    out = [o for o in result.output.splitlines() if o]
+
+    assert len(out) == 4
+
+
+def test_invalid_push(capfd: pytest.CaptureFixture[str]) -> None:
+    """Ensure _push_image() fails when provided with an invalid profile."""
+    profile_ = profile.Profile(logger, False, "amd64", "glibc", False, True, False)
+    profile_.profile = "asdjadnancalca"  # Garbage in
+
+    push_log = sut._push_image(profile=profile_).splitlines()
+    out = [o for o in capfd.readouterr()[1].splitlines() if o]
+
+    assert len(push_log) == 2
+    assert "error" in push_log[1]
+
+    assert len(out) == 1
+
+
+@pytest.mark.slow
```

Comment:
Unit tests should never be slow.  This should mock out the actual build elements and only test the logic of the top level behaviour.

Reply:
This actually seems to be an "integration test". Maybe we should make this a unit test instead and leave integration tests for another test suite?

Reply:
This is actually a system or end to end test if I recall correctly.  I'd say if we want a system test suite (there's some utility there) then we should definitely move it to another suite or mark it with a system tag that's not run by default.  If we feel that those tests get us something we can add them to a release workflow and have a good argument for moving to git flow sooner rather than later.